### PR TITLE
New test syntax with fqids

### PR DESCRIPTION
**Pull request by ostcar**
@normanjaeckel could you look over the new syntax?

@FinnStutzenstein This will test every field of each collection. It uses go generate to fetch all fields and test them.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -12,29 +11,42 @@ There are two types of tests:
...
+    attribute.
+
+  * `can_see`: A list of fields that the user is expected to see. This field is
+    the expected exact outout of the `RestrictFQFields` function. A value can also
+    be a fqid. In this case, all fields of the collection should be seen.
+
+  * `can_see_at_least`: Like `can_see` but does not throw an error, if more
+    fields are returned.
+
+  * `can_not_see`: A list of fields that the user is now allowed to see. A value
```
Comment (normanjaeckel) [inline tests/README.md]: Minor proposal: Add `can_not_see_at_least`

Comment (ostcar) [reply tests/README.md]: Other proposal:

can_see means "can see at least"
can_not_see means "can not see at least"
can_see_only or can_see_exact is the current can_see and the exact output of the restrict interface.

**Comment by ostcar** [comment]
@normanjaeckel Finally, I understood what you meant. This is the implementation. Now there are the parameters `can_see` and `can_not_see`. `can_see` is the exact output of `RestrictFQFields` and `can_not_see` is a shortcut for all fields from `fqfields` expect thes once.

**Comment by ostcar** [comment]
There is another but that I probably will not fix. This the yaml-lib that I use, I did not succeed to distinguish between nil and undefined. This means, the following yaml structures look the same for me: 

```yaml
fqfields:
- a
- b
can_see:
```
 
```yaml
fqfields:
- a
- b
```

Since a object without `can_see` means, that no test is run, the effect of both yaml-objects is, that no test is run. A workaround is to use one of these objects:

```yaml
fqfields:
- a
- b
can_see: []
```
 
```yaml
fqfields:
- a
- b
can_not_see:
- a
- b
```

**Pull request merged**